custom_components/better_thermostat/climate.py

```diff
@@ -910,18 +919,12 @@ async def _async_control_heating(self):
 				self._active = True
 				self.ignore_states = True
 				# Use the same precision and min and max as the TRV
-				if self.hass.states.get(self.heater_entity_id).attributes.get('target_temp_step') is not None:
+				if self.hass.states.get(self.heater_entity_id).attributes.get('target_temp_step') is not None and self.target_temperature_step is None:
```

Comment:
target_temperature_step is never None. It's either set by the config or the default value 0.5